frontend/packages/ceph-storage-plugin/src/components/create-backingstore-page/create-bs.tsx

```diff
@@ -62,6 +77,29 @@ const CreateBackingStoreForm: React.FC<CreateBackingStoreFormProps> = withHandle
     namespace,
   } = props;
 
+  const [isSubmit, setIsSubmit] = React.useState(false);
+  let isPhaseDefined = false;
+  const nooBaaBsResource: WatchK8sResource = React.useMemo(() => {
+    const storeName = isSubmit ? bsName : null;
+    return {
+      kind: referenceForModel(NooBaaBackingStoreModel),
+      namespaced: true,
+      isList: false,
+      name: storeName,
+      namespace,
+    };
+  }, [bsName, namespace, isSubmit]);
+  const [newNbsObj, newNbsLoaded, newNbsErr] = useK8sWatchResource<BackingStoreKind>(
+    nooBaaBsResource,
+  );
+
+  if (newNbsLoaded && !newNbsErr) {
+    isPhaseDefined = newNbsObj?.status?.phase !== undefined;
+  }
+  if (isSubmit && isPhaseDefined) {
+    close();
+  }
```

Comment:
you can move this inside useEffect so you don't need extra const isPhaseDefined

const backingStoreObject:  BackingStoreKind[] = useDeepCompareMemoize(newNbsObj, true);

```
React.useEffect(() => {
  if(isSubmit && !isPage){
   if(backingStoreObject && backingStoreObject?.status?.phase ===  'Ready')
      close()
  else if (backingStoreObject && backingStoreObject?.status?.phase ===  'Failed')
    // doSomething
  else:
      <EmptyState>
          <EmptyStateIcon icon={LoadingComponent} />
        </EmptyState>
 }
}, [backingStoreObject, isSubmit, newNbsLoaded, newNbsErr ]);
```

Reply:
https://github.com/openshift/console/blob/master/frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/create-block-pool-modal.tsx#L66